### ISSUE TITLE
implement Clone for NodeWeights

### ISSUE THREAD

**Issue by JonasOberhauser** [petgraph/petgraph#647]
### Summary

I think NodeWeights should implement Clone 

### Motivation

Sometimes I need to iterate over the crossproduct of some set of nodes. That would be really easy if I could clone the iterator over the set of nodes.

### Details

* What code needs to change? New traits? Which modules?

Maybe just need to add
```#[derive(Clone)]```
before https://docs.rs/petgraph/latest/src/petgraph/graph_impl/mod.rs.html#1771 . Haven't tried though.

**Comment by bluss** [comment]
Yes, makes sense. Needs custom but easy impl to avoid erroneous default clone bound on the N type parameter.
